### ISSUE TITLE
Rule: Fields should not be initialized to default values

### ISSUE THREAD

**Issue by dbolkensteyn** [SonarSource/sonarlint-vs#81]
Replaces http://jira.sonarsource.com/browse/LINTVS-98

A reference type field is automatically initialized to `null`, while a value type is initialized to the default value of the type. So initializing a field to its default value is useless. This initialization can happen both in a field initializer or in the constructor.
R# only reports on field initializers, and not constructor ones (it might not be that simple to recognize the latter).

```
class X
{
  public int field = 0; //Noncompliant, it is already 0, it has a minor performance impact that we reinitialize
  public object o = null; //Noncompliant (R# doesn't warn on this one)
}
```

R# ID: *RedundantDefaultFieldInitializer*

http://www.codeproject.com/Articles/10439/Gain-performance-by-not-initializing-variables
http://blog.codinghorror.com/for-best-results-dont-initialize-variables/

Implements http://jira.sonarsource.com/browse/RSPEC-3052

**Comment by dbolkensteyn** [comment]
@ganncamp : I've associated this to an existing rule: http://jira.sonarsource.com/browse/RSPEC-3052

**Comment by dbolkensteyn** [comment]
test: depends on https://github.com/dotnet/roslyn/issues/4021
